client/src/components/AdventModal.js

```diff
@@ -1,81 +1,32 @@
-import React from 'react';
+import Modal from 'react-modal';
 import styled from 'styled-components/macro';
-import data from '../data.json';
-
-import { makeStyles } from '@material-ui/core/styles';
-import Modal from '@material-ui/core/Modal';
-
-function rand() {
-  return Math.round(Math.random() * 20) - 10;
-}
-
-function getModalStyle() {
-  const top = 50 + rand();
-  const left = 50 + rand();
-
-  return {
-    top: `${top}%`,
-    left: `${left}%`,
-    transform: `translate(-${top}%, -${left}%)`,
-  };
-}
-
-const useStyles = makeStyles((theme) => ({
-  paper: {
-    position: 'absolute',
-    width: 150,
-    backgroundColor: theme.palette.background.paper,
-    border: '2px solid #000',
-    boxShadow: theme.shadows[5],
-    padding: theme.spacing(2, 4, 3),
-  },
-}));
-
-export default function AdventModal() {
-  const classes = useStyles();
-  const [modalStyle] = React.useState(getModalStyle);
-  const [open, setOpen] = React.useState(false);
-
-  const handleOpen = () => {
-    setOpen(true);
-  };
-
-  const handleClose = () => {
-    setOpen(false);
-  };
 
+export default function AdventModal({ advDayContent, isOpen, onClose }) {
   const body = (
-    <div>
-      <h2 id="simple-modal-title">Text in a modal</h2>
-      <p id="simple-modal-description">
-        Duis mollis, est non commodo luctus, nisi erat porttitor ligula.
-      </p>
-      <AdventModal />
-    </div>
+    <ModalStyle>
+      <CloseButton onClick={onClose}>x</CloseButton>
+
+      <h2>{advDayContent.day}. Dez. </h2>
+      <p> Wir verlosen heute: </p>
+      {advDayContent.text}
+    </ModalStyle>
   );
 
   return (
-    <div>
-      <Button type="button" onClick={handleOpen}>
-        Open Modal
-      </Button>
-
-      <Modal
-        open={open}
-        onClose={handleClose}
-        aria-labelledby="simple-modal-title"
-        aria-describedby="simple-modal-description"
-      >
-        {body}
-      </Modal>
-    </div>
+    <Modal isOpen={isOpen} onClose={onClose}>
```

Comment:
not sure how Modal works but could you write it like that? Maybe better to read.

`
  return (
    <Modal isOpen={isOpen} onClose={onClose}>
      <ModalStyle>
      <CloseButton onClick={onClose}>x</CloseButton>

      <h2>{advDayContent.day}. Dez. </h2>
      <p> Wir verlosen heute: </p>
      {advDayContent.text}
    </ModalStyle>
    </Modal>
  );
`

Reply:
... or rename {body} in {ModalContent} to get it right what it means. To have it in a variable makes it a little bit easier to reuse it in other components. But sure, one can write the code as you suggested.